### ISSUE TITLE
Rule proposal: require-emit-validator

### ISSUE THREAD

**Issue by g-plane** [vuejs/eslint-plugin-vue#1482]
**Please describe what the rule should do:**

Enforce detailed `emits` option with types.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```ts
// bad
export default defineComponent({
  emits: ['update'],
})

// good
export default defineComponent({
  emits: {
    update: (payload: Payload) => typeof payload === 'object',
  },
})
```

**Additional context**

- Declaring types of `emits` can provide better TypeScript inference.
- This rule is similar to [require-prop-types](https://eslint.vuejs.org/rules/require-prop-types.html).

**Comment by ota-meshi** [comment]
Thank you for suggesting this rule. I like this rule!
